chanzuckerberg/single-cell-data-portal
# feat: generate and log request_ids for the backend API

- #TICKET_NUMBER

### Reviewers
**Functional:** @ebezzi 

**Readability:** @nayib-jose-gloria 

---

Adding a request_id to all requests made to the backend API. This will allow us to filter logs per request and will greatly help with debugging backend errors.

## Changes
- request id's are uuid4 format
- Add request_id at the start of a request
- Return the request_id in the response
- Log request_id in all logs with that request context
- tested that X-Request-Id exists

## QA steps (optional)
- Added test case to verify a request_id is generated and returned
- Checked logs to see that request_id is populated matching per request.
- 
## Notes for Reviewer
